packages/cdktf/lib/backends/cos-backend.ts

```diff
@@ -16,13 +16,14 @@ export class CosBackend extends TerraformBackend {
   }
 
   public getRemoteStateDataSource(
-    _scope: Construct,
-    _name: string,
+    scope: Construct,
+    name: string,
     _fromStack: string
   ): TerraformRemoteState {
-    throw new Error(
-      "The CosBackend does not yet implement cross stack references"
-    );
+    return new DataTerraformRemoteStateCos(scope, name, {
+      ...this.props,
+      workspace: "${terraform.workspace}",
```

Comment:
If there is any case where one would want to overwrite `workspace` there is no chance now since the spread comes before the `workspace` value. If that case doesn't exist we have no issue, if it does we should put the spread in as last element so it overwrites previous values.

Reply:
This method of overwriting `workspace` would be very obscure and only work in Typescript since the type of the `props` variable doesn't actually include `workspace`. If we wanted to provide overwrite support, we'd probably want it as an option at the stack level. I see that as more of a follow up task (assuming there is even demand for it).

For context, I added the workspace in since I currently have environments separated by workspaces in Terraform and am considering keep it that way when migrating to cdktf. I'm not certain that I'll go that way, but nice to have the option. I see two dimensions of stacks: one for different parts of the application and one for different environments. Since different environments are typically deployed at different times, switching Terraform workspace in-between seems viable.

Reply:
Ah that makes sense, so it's not related but adjacent :)